cmake/kokkos_arch.cmake

```diff
@@ -300,6 +302,20 @@ IF (KOKKOS_ARCH_A64FX)
   )
 ENDIF()
 
+IF (KOKKOS_ARCH_ARMV9_GRACE)
+  SET(KOKKOS_ARCH_ARM_NEON ON)
+  check_cxx_compiler_flag("-mcpu=neoverse-n2" COMPILER_SUPPORTS_NEOVERSE_N2)
+  check_cxx_compiler_flag("-msve-vector-bits=128" COMPILER_SUPPORTS_SVE_VECTOR_BITS)
+  IF (COMPILER_SUPPORTS_NEOVERSE_N2 AND COMPILER_SUPPORTS_SVE_VECTOR_BITS)
+    COMPILER_SPECIFIC_FLAGS(
+      COMPILER_ID KOKKOS_CXX_HOST_COMPILER_ID
+      DEFAULT -mcpu=neoverse-n2 -msve-vector-bits=128
+    )
+  ELSE()
+    MESSAGE(WARNING "Compiler does not support ARMv9 Grace architecture")
+  ENDIF()
+ENDIF()
```

Comment:
Looks like we won't get this pull request in for the release when defining a new host architecture.
```suggestion
```

Reply:
I debugged it and it is fixed now. We can not use neo vectorization on `ARMV9_GRACE ` if we enable CUDA with nvcc. We missed some checks for that and I added them. Let me clean the code and push it.

Reply:
@masterleinad my last commit fixes the compilation issue.